buildspecs/buildspec-build-invoke.yml

```diff
@@ -0,0 +1,16 @@
+version: 0.2
+
+phases:
+  install:
+    runtime-versions:
+      docker: 19
+
+    commands:
+      - pip install --upgrade pip aws-sam-cli
```

Comment:
Had same thought when I tried. brew is not pre-installed, and installing brew is quite slow. Upgrading it using pip is faster, and sam cli is preinstalled using pip.

https://github.com/aws/aws-codebuild-docker-images/blob/ef3136e84a658442572cacf8f2686f970d7be5c3/al2/x86_64/standard/3.0/Dockerfile#L326

Reply:
I know installing brew on linux is really slow, can we test one more time with this command if this executes faster?
`HOMEBREW_NO_AUTO_UPDATE=1 brew install <formula>`

Otherwise I am OK to move forward with this

Reply:
I spent half an hour trying to install `brew` to codebuild image. I couldn't get it to work 😢 

It is brew itself takes a long time to install. I tested on a similar-size EC2 instance and it took 3.5 mins to install brew alone. 

```
$ time /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
real3m37.988s
user6m2.794s
sys0m5.932s
```

homebrew requires non-root to run the installation script, which invoke `sudo` at some point, which made it very complex to configure the environment:

Attaching my failed attempt, I don't think it is good to use brew due to these...

```
version: 0.2

run-as: codebuild-user

phases:
    install:
        run-as: root
        commands:
            - yum install sudo -y
            - echo "codebuild-user   ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers

    pre_build:
        commands:
            - which sudo
            - sudo ls -l /
            - /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
            - brew tap aws/tap
            - brew install aws-sam-cli

    build:
        commands:
            - PATH=~/.linuxbrew/bin:$PATH
            - sam --version
```

Reply:
I am fine without using linuxbrew, however I think you are pretty close :)

I think you just need to update path before you call `brew tap`. Check [this example](https://github.com/aws/homebrew-tap/blob/master/appveyor.yml#L25-L26)

Reply:
Brew was not installed successfully actually:

<img width="1946" alt="image" src="https://user-images.githubusercontent.com/3804518/101545534-3ee68000-395c-11eb-89ae-d5733a3e8e0e.png">


[Container] 2020/12/08 21:19:37 Running command /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
--